draft/spec/index.html

```diff
@@ -868,6 +868,34 @@ <h2>Extensions</h2>
             </p>
         </blockquote>
     </section>
+    
+    <section id="filesystem-features">
+        <h2>Filesystem features</h2>
+        <p>
+            In order to maximise the compatibility of the OCFL with different filesystems, and thus improve the 
+            portability of OCFL Objects betwwen different systems, some restrictions on the use of certain 
+            filesystem features are necessary. If the preservation of non-OCFL-compliant features is required then
+            data MUST be wrappered in a suitable disk or filesystem image format which OCFL can treat as a regular
+            file.
+        </p>
+        <ol>
+          <li>The use of JSON for the inventory means that storage paths are stored in as Unicode. Therefore, the 
+              filesystem hosting an OCFL Storage Root SHOULD support Unicode file- and path-names. If not, there
+              MUST be a consistent, symmetric mapping between the character set used for filesystem paths and
```

Comment:
I'm not quite sure I understand what this requirement is saying.  If the storage filesystem does not support unicode, then isn't it simply the case that the "content paths" (which I believe is the new terminology for what was formerly known as "existing file path") are subject to the limitations of the local filesystem, and therefore cannot contain unicode on filesystems that don't support it?  

Additionally, it's not entirely clear where/how this mapping occurs.  When transferring an OCFL object between systems with different filesystem naming capabilities, is this mapping referring to software that might need to re-write the filenames and update the `inventory.json` files to account for local filesystem limitations?  i.e. _there MUST be a consistent, symmetric mapping..._ is placing constraints on the process of migrating an OCFL objects between filesystems of different capabilities?

Reply:
I think the issue we've identified is where objects traverse filesystems that may have different character encodings for their filesystems. A filesystem can support Unicode, but not in the same way that OTHER filesystems support it.

HFS+, for example, treats Unicode in a different way from most other modern filesystems, using a modified variant of NFD. 

https://en.wikipedia.org/wiki/HFS_Plus#Design

APFS, on the other hand, does not use any Unicode normalization:

https://mjtsai.com/blog/2017/03/24/apfss-bag-of-bytes-filenames/

This can cause problems:

https://eclecticlight.co/2017/04/06/apfs-is-currently-unusable-with-most-non-english-languages/

APFS isn't the only one:

https://gist.github.com/Kwpolska/0c63a22f65efe9c90607a826aad99d1f

I think fundamentally we just don't have enough data to make a real recommendation here, so we're hedging our bets.

Reply:
I believe I understand the motivating issue.  The problem is that the language is not clear as to what "consistent, symmetric mapping" means.  Where does the mapping occur, who is performing the mapping, and what implication does it have for the contents of the `inventory.json` file if any?

To use an HFS+ example, the filesystem itself performs a "mapping" (of é to the three bytes 65 cc 81, which may not be equivalent to the bytes used for é in the inventory storage paths.).  Is this clause just saying that it is OK for the to use OCFL with filesystems that do that?  Or is it allowing (or requiring) OCFL software to implement any sort of workaround it wants in order to deal with `inventory.json` and filesystem path diffferences, as long as it involves a "a consistent, symmetric mapping"?  

Absent the mapping clause, my interpretation is that the storage paths used in the `inventory.json` file would be subject to the local filesystem costraints.  i.e. if your FS requires é to be `65 cc 81`, those bytes must be in the storage path in the inventory, no exceptions.  I thought this is where the flexibity afforded by logical paths comes in.  If the desired name of the file has é as `c3 a9`, it's fine to have that normalization in the logical file path, but if you happen to be using HFS+ to persist OCFL, your hands are tied to using  `65 cc 81` in the storage path.

Reply:
I think we're trying to deal with a case where an OCFL object written on Filesystem A using `65 cc 81` writes an inventory, and then the whole lot gets moved to Filesystem B, where the file is understood as `c3 a9`. In this case, as I understand it, the inventory would contain the bytes `65 cc 81` (because it doesn't get rewritten) but a client attempting to look that path up on the filesystem wouldn't find it. This might happen if, for example, you copy a file from a local filesystem to an Object Store.

Like I said, we don't really have a whole lot of use cases to know if this is an actual worry, or just one that we're imagining.

Reply:
I believe filesystems that do not preserve case (i.e. the FS operations normalize lowercase characters to uppercase) are an extreme example of the same phenomenon, and may be more tangible https://github.com/OCFL/spec/issues/285

On Windows, it's even worse.  Characters like `:`,`<`,`?`,`*` are prohibited entirely, as are character sequences like `NUL`.   The filesystem throws an error, rather than re-mapping a character automatically.  The mapping clause might be interpreted as allowing an implementation to re-name the files using its own convention, as long as it's consistent with the rules it applies to do so.  Is this accurate?  

I worry that there is a point at which flexibility in the name of portability can complicate the archival characteristics of OCFL.

Reply:
It's one of the reasons why we're recommending that if you do want to do something with older filesystems, ISO or tar formats may be useful. Especially when it comes to archiving full disk images with empty directories.

Reply:
Fair enough.  As an outsider, it's easier to understand the idea that portability might not be possible across filesystems, and there is nothing that can be done about that aside from judicious choices of storage paths (file names) in the first place, or wrapping it in an archival storage format like a tar.  So the first paragraph of this PR is fine, it's just that the normative remediation for filesystem path inconsistencies (due to character encoding, support for ASCII case, unsupported characters, etc) is something else completely: a mapping of some sort, the nature of which is undefined.  Why wouldn't path issues fall under the purvue of "if portability problems arise, you need to wrap" as well?

Reply:
...This should probably be dealt with in the Implementation Notes rather than here.